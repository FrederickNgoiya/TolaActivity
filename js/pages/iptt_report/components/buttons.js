import React from 'react';
import { inject, observer } from 'mobx-react';
import Popover from 'react-simple-popover';

@inject('labels', 'rootStore')
class PinPopover extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            reportName: '',
            sending: false,
            sent: false
        };
    }
    handleChange = (e) => {
        this.setState({reportName: e.target.value});
    }
    isDisabled = () => {
        return !this.props.rootStore.pinData || !this.state.reportName;
    }
    handleClick = () => {
        //send this.props.rootStore.pinUrl as data to this.props.labels.pin.creatUrl
        //with data "name" as this.state.reportName
        this.setState({sending: true});
        $.ajax({
            type: "POST",
            url: this.props.labels.pin.createUrl,
            data: {name: this.state.reportName, ...this.props.rootStore.pinData },
            success: () => {this.setState({sending:false, sent: true}); },
            error: () => {console.log("AJAX ERROR");}
        });
    }
    render() {
        return (
            <React.Fragment>
                { this.state.sent
                    ? <div className="form-group">
                        <p>
                            <span>{ this.props.labels.pin.successMsg }</span>
                        </p>
                        <p>
                           <a href={ this.props.rootStore.programPageUrl }>
                                { this.props.labels.pin.successLink }
                           </a>
                        </p>
                      </div>
                    : <React.Fragment>
                        <div className="form-group">
                            <label className="text-uppercase">
                               { this.props.labels.pin.reportName }
                            </label>
                            <input type="text" className="form-control"
                                 value={ this.state.reportName }
                                 onChange={ this.handleChange }
                                 disabled={ this.state.sending }/>
                        </div>
                        { this.state.sending
                            ? <div className="btn btn-outline-primary" disabled>
                                <img src='/static/img/ajax-loader.gif' />&nbsp;
                                    { this.props.labels.loading }
                              </div>
                            : <button type="button"
                                  onClick={ this.handleClick }
                                  disabled={ this.isDisabled() }
                                  className="btn btn-primary">
                                      { this.props.labels.pin.submitButton }
                              </button>
                        }
                        </React.Fragment>
                        
                }
            </React.Fragment>
        );
    }
}

@inject('labels')
export class PinButton extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false
        };
    }
    
    handleClick(e) {
        this.setState({open: !this.state.open});
    }
    
    handleClose(e) {
        this.setState({open: false});
    }
    
    render() {
        return (
            <React.Fragment>
                <div
                    className="btn btn-sm btn-secondary"
                    ref="target"
                    onClick={ this.handleClick.bind(this) }>
                <i className="fas fa-thumbtack"></i> { this.props.labels.pin.buttonLabel }
                </div>
                <Popover
                    placement='bottom'
                    style={ {width: 'auto' } }
                    target={ this.refs.target }
                    show={ this.state.open }
                    onHide={ this.handleClose.bind(this) } >
                    <PinPopover />
                </Popover>
            </React.Fragment>
        );
    }
}

export const ExcelButton = inject('labels')(({ labels }) => {
        return <button type="button" className="btn btn-sm btn-secondary">
                    <i className="fas fa-download"></i> { labels.excel }
                </button>;
            
    }
);