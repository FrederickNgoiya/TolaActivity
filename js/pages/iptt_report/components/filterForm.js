import React from 'react';
import { inject, observer } from 'mobx-react';
import { computed } from 'mobx';
import Select from 'react-select';

const programOptions = [
    { value: 1, label: 'program 1'},
    { value: 2, label: 'program 2'}
];
const selectedProgram = {value: 1, label: 'program 1'};

const IPTTSelectWrapper = (props) => {
    return <div className="form-row mb-3">
                <label className="col-form-label text-uppercase">
                    { props.label }
                </label>
                { props.children }
            </div>;
}


@inject('labels', 'rootStore')
@observer
class ProgramSelect extends React.Component {
    selectProgram = (selected) => {
        this.props.rootStore.setProgramId(selected.value);
    }
    render() {
        return <IPTTSelectWrapper label={ this.props.labels.programSelect }>
                    <Select options={ this.props.rootStore.programOptions }
                            value={ this.props.rootStore.selectedProgramOption }
                            onChange={ this.selectProgram }
                            className="iptt-react-select" />
               </IPTTSelectWrapper>;
    }
}

@inject('labels', 'rootStore')
@observer
class PeriodSelect extends React.Component {
    selectFrequency = (selected) => {
        this.props.rootStore.setFrequencyId(selected.value);
    }
    render() {
        const label = this.props.rootStore.isTVA
                ? this.props.labels.periodSelect.tva
                : this.props.labels.periodSelect.timeperiods;
        return <IPTTSelectWrapper label={ label }>
                    <Select options={ this.props.rootStore.frequencyOptions }
                            value={ this.props.rootStore.selectedFrequencyOption }
                            onChange={ this.selectFrequency }
                            className="iptt-react-select" />
               </IPTTSelectWrapper>;
    }
}

@inject('rootStore')
@observer
class DateSelect extends React.Component {
    @computed get options() {
        if (this.props.rootStore.selectedFrequencyId == 7) {
            return Object.entries(this.props.rootStore.periodOptions).map(
                ([optgroupLabel, options], index) => {
                    return <optgroup label={optgroupLabel} key={index}>
                        {options.map(
                            (option) => <option value={option.value} key={option.value}>{option.label}</option>
                        )}
                    </optgroup>;
                }
            );
        } else {
            return this.props.rootStore.periodOptions.map(
                (option) => <option value={option.value} key={option.value}>{option.label}</option>
            );
        }
    }
    render() {
        return <IPTTSelectWrapper label={ this.props.label }>
                    <select className="form-control"
                            value={ this.props.value }
                            onChange={ this.props.onChange }
                            disabled={ this.props.rootStore.selectedFrequencyId == 2 || this.props.rootStore.selectedFrequencyId == 1 }>
                        { this.options }
                    </select>
                </IPTTSelectWrapper>;
    }
}

const StartDateSelect = inject('labels', 'rootStore')(
    observer(({labels, rootStore}) => {
        const selectHandler = (e) => { rootStore.setStartPeriod(e.target.value); }
        return <DateSelect label={ labels.startPeriod }
                           value={ rootStore.startPeriod }
                           onChange={ selectHandler } />
    })
);

const EndDateSelect = inject('labels', 'rootStore')(
    observer(({labels, rootStore}) => {
        const selectHandler = (e) => { rootStore.setEndPeriod(e.target.value); }
        return <DateSelect label={ labels.endPeriod }
                           value={ rootStore.endPeriod }
                           onChange={ selectHandler } />
    })
);

@inject('labels', 'rootStore')
@observer
class TimeFrameRadio extends React.Component {
    checkMostRecent = () => {
        //default value of 2 in case of clicking "most recent" radio box - default behavior
        this.props.rootStore.setMostRecent(2);
    }
    updateMostRecentCount = (e) => {
        this.props.rootStore.setMostRecent(e.target.value);
    }
    render() {
        return <div className="form-row mb-3">
                    <div className="col-sm-4">
                        <div className="form-check form-check-inline pt-1">
                            <span className="form-check-input">
                                <input type="radio"
                                       checked={ this.props.rootStore.showAll }
                                       disabled={ !this.props.rootStore.timeframeEnabled }
                                       onChange={ this.props.rootStore.setShowAll }
                                       />
                            </span>
                            <label className="form-check-label">
                                { this.props.labels.showAll }
                            </label>
                        </div>
                    </div>
                    <div className="col-sm-4 p-0">
                        <div className="form-check form-check-inline pt-1">
                            <span className="form-check-input">
                                <input type="radio"
                                       checked={ this.props.rootStore.mostRecent !== null }
                                       disabled={ !this.props.rootStore.timeframeEnabled }
                                       onChange={ this.checkMostRecent }
                                       />
                            </span>
                            <label className="form-check-label">
                                { this.props.labels.mostRecent}
                            </label>
                        </div>
                    </div>
                    <div className="col-sm-4">
                        <input type="number" className="form-control"
                               value={ this.props.rootStore.mostRecent || ''}
                               disabled={ !this.props.rootStore.timeframeEnabled }
                               onChange={ this.updateMostRecentCount }
                               />
                    </div>
               </div>;
    }
}

const IPTTMultiSelectWrapper = (props) => {
    return <div className="form-row mb-2">
                <label className="col-form-label text-uppercase">
                    { props.label }
                </label>
                { props.children }
            </div>;
}

@inject('labels', 'rootStore')
@observer
class LevelSelect extends React.Component {
    getOptions = () => {
        return this.props.rootStore.selectedProgram.reportLevels.map(
            level => ({value: level, label: level})
        );
    }
    updateLevelFilters = (selected) => {
        this.props.rootStore.levelFilters = selected;
    }
    render() {
        return <IPTTMultiSelectWrapper label={this.props.labels.levelSelect}>
                    <Select options={this.getOptions()}
                            isMulti={true}
                            value={ this.props.rootStore.levelFilters }
                            onChange={ this.updateLevelFilters }
                            className="iptt-react-select"/>
                </IPTTMultiSelectWrapper>;
    }
}

@inject('labels', 'rootStore')
@observer
class SiteSelect extends React.Component {
    updateSiteFilters = (selected) => {
        this.props.rootStore.siteFilters = selected;
    }
    render() {
        return <IPTTMultiSelectWrapper label={this.props.labels.siteSelect}>
                    <Select options={this.props.rootStore.selectedProgram.reportSites}
                            isMulti={ true }
                            value={ this.props.rootStore.siteFilters }
                            onChange={ this.updateSiteFilters }
                            className="iptt-react-select" />
                </IPTTMultiSelectWrapper>
    }
}

@inject('labels', 'rootStore')
@observer
class TypeSelect extends React.Component {
    updateTypeFilters = (selected) => {
        this.props.rootStore.typeFilters = selected;
    }
    render() {
        return <IPTTMultiSelectWrapper label={this.props.labels.typeSelect}>
                    <Select options={this.props.rootStore.selectedProgram.reportTypes}
                            isMulti={ true }
                            value={ this.props.rootStore.typeFilters }
                            onChange={ this.updateTypeFilters }
                            className="iptt-react-select" />
                </IPTTMultiSelectWrapper>
    }
}

@inject('labels', 'rootStore')
@observer
class SectorSelect extends React.Component {
    updateSectorFilters = (selected) => {
        this.props.rootStore.sectorFilters = selected;
    }
    render() {
        return <IPTTMultiSelectWrapper label={this.props.labels.sectorSelect}>
                    <Select options={this.props.rootStore.selectedProgram.reportSectors}
                            isMulti={ true }
                            value={ this.props.rootStore.sectorFilters }
                            onChange={ this.updateSectorFilters }
                            className="iptt-react-select" />
                </IPTTMultiSelectWrapper>
    }
}


export const IPTTFilterForm = inject('labels')(
    observer(({labels}) => {
        return <nav id="id_iptt_report_filter">
                    <div className="p-3" id="filter-top">
                        <h3 className="filter-title text-title-case">{ labels.filterTitle }</h3>
                        <ProgramSelect />
                        <PeriodSelect />
                        <TimeFrameRadio />
                        <StartDateSelect />
                        <EndDateSelect />
                    </div>
                    <div id="filter-middle" className="p-3">
                        <LevelSelect />
                        <SiteSelect />
                        <TypeSelect />
                        <SectorSelect />
                    </div>
                </nav>;
    })
);