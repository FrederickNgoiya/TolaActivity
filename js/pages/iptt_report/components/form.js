import React from 'react';
import {
    ProgramSelect,
    PeriodSelect,
    ShowAllSelect,
    MostRecentSelect,
    MostRecentCount,
    StartPeriodSelect,
    EndPeriodSelect,
    LevelSelect,
    TypeSelect,
    SectorSelect,
    SiteSelect,
    IndicatorSelect
} from '../../iptt_shared/components/form';

export class IPTTFilterForm extends React.Component {
    render() {
        return <React.Fragment>
                <div id="filter-top" className="p-3">
                    <h3 className="filter-title">{this.props.uiStore.labels.filterTitle}</h3>
                    <ProgramSelect {...this.props}
                        classNames="form-control" outerClassNames="form-row mb-3"
                        labelClassNames="col-form-label" />
                    <PeriodSelect {...this.props}
                        classNames="form-control" outerClassNames="form-row mb-3"
                        labelClassNames="col-form-label" />
                    <div className="form-row mb-3">
                        <div className="col-sm-4">
                            <ShowAllSelect {...this.props} outerClassNames="pt-1" />
                        </div>
                        <div className="col-sm-4 p-0">
                            <MostRecentSelect {...this.props} outerClassNames="pt-1" />
                        </div>
                        <div className="col-sm-4">
                            <MostRecentCount {...this.props} />
                        </div>
                    </div>
                    <div className="form-row mb-3">
                        <StartPeriodSelect {...this.props} />
                    </div>
                    <div className="form-row mb-3">
                        <EndPeriodSelect {...this.props} />
                    </div>
                </div>
                <div id="filter-middle" className="p-3">
                    <LevelSelect {...this.props} outerClassNames="form-group mb-2"
                                labelClassNames="col-form-label text-uppercase" classNames="form-control" />
                    <TypeSelect {...this.props} outerClassNames="form-group mb-2"
                                labelClassNames="col-form-label text-uppercase" classNames="form-control" />
                    <SectorSelect {...this.props} outerClassNames="form-group mb-2"
                                labelClassNames="col-form-label text-uppercase" classNames="form-control" />
                    <SiteSelect {...this.props} outerClassNames="form-group mb-2"
                                labelClassNames="col-form-label text-uppercase" classNames="form-control" />
                    <IndicatorSelect {...this.props} outerClassNames="form-group mb-2"
                                labelClassNames="col-form-label text-uppercase" classNames="form-control" />
                </div>
                <div id="filter-bottom" className="d-flex justify-content-between no-gutters p-3">
                    <div className="col pr-2">
                        <button onClick={this.props.uiStore.applyFilters} className="btn btn-block btn-primary">
                            {this.props.uiStore.labels.applyButton}
                        </button>
                    </div>
                    <div className="col pl-2">
                        <button onClick={this.props.uiStore.resetAll} className="btn btn-block btn-inverse">
                            {this.props.uiStore.labels.resetButton}
                        </button>
                    </div>
                </div>
                </React.Fragment>
    }
}