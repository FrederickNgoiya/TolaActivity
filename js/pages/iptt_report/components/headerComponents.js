import React from 'react';
import { observer, inject } from 'mobx-react';


const SubheadCell = (props) => {
    let tdStyle = {};
    if (props.minWidth) {
        tdStyle.minWidth = props.minWidth;
    }
    return <th scope="col" className={"align-bottom text-uppercase " + (props.classes || '')}
            style={ tdStyle }>{ props.cellText }</th>;
}

@inject('labels', 'rootStore')
@observer
class IPTTSubheadRow extends React.Component {
    getDateRangeHeaders = () => {
        let headers = [];
        let periods = this.props.rootStore.selectedPeriods;
        if (!periods) {
            return headers;
        } else if (this.props.rootStore.isTVA) {
            periods.forEach(
                (period, index) => {
                    headers.push(<SubheadCell classes="text-right" minWidth="110px" key={ index + 'a' }
                            cellText={ this.props.labels.columnHeaders.target} />);
                    headers.push(<SubheadCell classes="text-right" minWidth="110px" key={ index + 'b' }
                            cellText={ this.props.labels.columnHeaders.actual} />);
                    headers.push(<SubheadCell classes="text-right" minWidth="110px" key={ index + 'c' }
                            cellText={ this.props.labels.columnHeaders.met} />);
                }
            );
        } else {
            periods.forEach(
                (period, index) => {
                    headers.push(<SubheadCell classes="text-right" key={ index }
                            cellText={ this.props.labels.columnHeaders.actual} />);
                }
            );
        }
        return headers;
    }
    render() {
        let colLabels = this.props.labels.columnHeaders;
        return <tr>
                    <SubheadCell minWidth='80px' cellText={ colLabels.number } />
                    <SubheadCell minWidth='600px' classes='td-no-side-borders'
                                 cellText={ colLabels.indicator } />
                    <SubheadCell classes="td-no-side-borders" />
                    <SubheadCell minWidth='90px' cellText={ colLabels.level } />
                    <SubheadCell minWidth='250px' cellText={ colLabels.uom } />
                    <SubheadCell cellText={ colLabels.change } />
                    <SubheadCell minWidth='130px' cellText={ colLabels.cumulative } />
                    <SubheadCell minWidth='50px' cellText={ colLabels.numType } />
                    <SubheadCell cellText={ colLabels.baseline } />
                    <SubheadCell minWidth='110px' classes='text-right td-no-side-borders'
                        cellText={ colLabels.target } />
                    <SubheadCell minWidth='110px' classes='text-right td-no-side-borders'
                        cellText={ colLabels.actual } />
                    <SubheadCell minWidth='110px' classes='text-right td-no-side-borders'
                        cellText={ colLabels.met } />
                    { this.getDateRangeHeaders() }
               </tr>;
    }
}



export const IPTTTableHead = inject('labels', 'rootStore')(
    observer(
        ({ labels, rootStore }) => {
            let getDateHeaders = () => {
                let headers = [];
                let periods = rootStore.selectedPeriods;
                if (!periods) {
                    return headers;
                }
                periods.forEach(
                    (period, index) => {
                        index += parseInt(rootStore.startPeriod);
                        let label = rootStore.getPeriodLabel(period, index);
                        headers.push(
                            <td scope="col" colSpan={ rootStore.isTVA ? '3' : '1'}
                                className={"text-center lopCols text-nowrap align-bottom" + (rootStore.selectedFrequencyId == 7 ? " text-uppercase" : "")} 
                                key={ index }> { label.title }<br /><small>{ label.subtitle }</small>
                            </td>
                        );
                    }
                );
                return headers;
            }
            return <thead className="thead-light">
                        <tr>
                            <td colSpan="9" id="id_td_iptt_program_name"
                                className="lopCols align-bottom pt-2"> 
                                <h5 className="m-0">
                                    <strong>{ rootStore.selectedProgram.name }</strong>
                                </h5>
                            </td>
                            <td scope="colgroup" colSpan="3"
                             className="text-center align-bottom text-uppercase">
                             { labels.columnHeaders.lop }
                             </td>
                             { getDateHeaders() }
                        </tr>
                        <IPTTSubheadRow />
                    </thead>
        }
    )
)