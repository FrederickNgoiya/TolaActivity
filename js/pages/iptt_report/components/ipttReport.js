import React from 'react';
import IPTTSidebar from './sidebar/sidebar';
import IPTTReportBody from './report/reportBody';

export default () => {
    return (
        <React.Fragment>
            <IPTTSidebar />
            <IPTTReportBody />
        </React.Fragment>
    );
}