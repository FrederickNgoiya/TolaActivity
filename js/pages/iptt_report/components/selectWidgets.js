import React from 'react';
import { inject, observer } from 'mobx-react';
import { computed } from 'mobx';
import Select from 'react-select';
import ReactMultiSelectCheckboxes from 'react-multiselect-checkboxes';
import { uniqueId } from '../../../formUtils';

const IPTTMultiSelectWrapper = (props) => {
    return <div className="form-row mb-2 iptt-react-select-row">
                <label htmlFor={props.id} className="col-form-label text-uppercase">
                    { props.label }
                </label>
                { props.children }
            </div>;
}


/**
 * styling element to replace OptGroup headings in react multiselect checkbox widgets - used for
 * LevelSelect (no optgroup label shown for the tiers group, no checkbox by other labels)
 */
const GroupHeading = (props) => {
    if (props.children == '') {
        return <div></div>;
    } else {
        return (
            <React.Fragment>
                <hr style={{ margin: '3px 0px 0px 0px' }} />
                <div style={{ textTransform: 'uppercase',
                              paddingLeft: '4px',
                              marginBottom: '2px'}}>
                    { props.children }
                </div>
            </React.Fragment>
            );
    }
}


export class IPTTMultiselectCheckboxWrapper extends React.Component {
    constructor(props) {
        super(props);
        this._id = uniqueId('multiselect');
    }
    get options() {
        return [];
    }
    get value() {
        return {value: null, label: ''};
    }
    get label() {
        return '';
    }
    getProps = () => {
        if (this.options.length == 0) {
            return {
                getDropdownButtonLabel: () => this.props.labels.noOptionsSelect,
                isDisabled: true,
                menuIsOpen: false,
                options: [],
                id: this._id,
            };
        }
        return {
            isMulti: true,
            id: this._id,
            options: this.options,
            getDropdownButtonLabel: (_ref) => {
                if (!_ref.value) {
                    return this.props.labels.emptySelect;
                }
                if (Array.isArray(_ref.value)) {
                    if (_ref.value.length == 0) {
                        return this.props.labels.emptySelect;
                    }
                    if (_ref.value.length == 1) {
                        return _ref.value[0].label;
                    }
                    return "".concat(_ref.value.length, " ", this.props.labels.selected);
                }
                return _ref.value.label;
            }
        };       
    }
    getBaseStyles = () => ({
        dropdownButton: base => this.options.length == 0
                                ? { ...base, backgroundColor: '#E5E6E8', background: '' }
                                : base,
        option: (provided, state) => ({
                ...provided,
                padding: '1px 12px',
                display: 'inline-block'
            }),
        container: (provided, state) => ({
                ...provided,
                backgroundColor: '#f5f5f5'
            }),
    });
    
    render() {
        const formatOptionLabel = (props) => {
            return <div style={{ display: "inline-block" , float: "right", width: "90%"}}>{props.label}</div>;
        }
        return <IPTTMultiSelectWrapper id={ this._id} label={ this.label }>
                    <ReactMultiSelectCheckboxes
                            styles={ this.getBaseStyles() }
                            formatOptionLabel={ formatOptionLabel }
                            components={{ GroupHeading }}
                            value={ this.value }
                            { ...this.getProps() }
                            onChange={ this.onChange } />
                </IPTTMultiSelectWrapper>;
    }
}



export const IPTTSelectWrapper = (props) => {
    return <div className="form-row mb-3">
                <label htmlFor={ props.id } className="col-form-label text-uppercase">
                    { props.label }
                </label>
                { props.children }
            </div>;
}

export class IPTTSelect extends React.Component {
    constructor(props) {
        super(props);
        this._id = uniqueId('react-select');
    }
    render() {
        return (
            <IPTTSelectWrapper id={ this._id } label={ this.label }>
                <Select {...this.selectOptions} id={ this._id } className="iptt-react-select" />
            </IPTTSelectWrapper>
        );
    }
}


@inject('rootStore')
@observer
export class DateSelect extends React.Component {
    constructor(props) {
        super(props);
        this._id = uniqueId('date-select');
    }
    @computed get options() {
        if (this.props.rootStore.selectedFrequencyId == 7) {
            return Object.entries(this.props.rootStore.periodOptions).map(
                ([optgroupLabel, options], index) => {
                    return <optgroup label={optgroupLabel} key={index}>
                        {options.map(
                            (option) => <option value={option.value} key={option.value}>{option.label}</option>
                        )}
                    </optgroup>;
                }
            );
        } else {
            return this.props.rootStore.periodOptions.map(
                (option) => <option value={option.value} key={option.value}>{option.label}</option>
            );
        }
    }
    render() {
        return <IPTTSelectWrapper id={ this._id }label={ this.props.label }>
                    <select className="form-control"
                            id={ this._id }
                            value={ this.props.value }
                            onChange={ this.props.onChange }
                            disabled={ this.props.rootStore.selectedFrequencyId == 2 || this.props.rootStore.selectedFrequencyId == 1 }>
                        { this.options }
                    </select>
                </IPTTSelectWrapper>;
    }
}