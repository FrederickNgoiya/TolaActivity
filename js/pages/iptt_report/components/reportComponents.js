import React from 'react';
import { observer, inject } from 'mobx-react';
import { IPTTTableHead } from './headerComponents';


const EMPTY_CELL = 'â€”';


const IndicatorTD = observer((props) => {
    return <td className="td-no-side-borders" {...props}>{props.children}</td>;
});


const TVAValue = observer(({ target, value, isPercent }) => {
    let percentText = (value && target && target != 0) ? String(Math.round(value / target * 1000) / 10) + '%' : EMPTY_CELL;
    let valueText = value ? String(Math.round(value)) + (isPercent ? '%' : '') : EMPTY_CELL;
    let targetText = target ? String(Math.round(target)) + (isPercent ? '%' : '') : EMPTY_CELL;
    return <React.Fragment>
        <td align="right">{targetText}</td>
        <td align="right">{valueText}</td>
        <td align="right">{percentText}</td>
    </React.Fragment>
});

@inject('rootStore')
@observer
class IndicatorRow extends React.Component {
    getPeriodValues = () => {
        if (!this.props.indicator.indicatorData) {
            return <td key={this.props.index}>Loading</td>;
        } else if (this.props.rootStore.isTVA) {
            return this.props.indicator.indicatorData.map(
                (values, index) => <TVAValue key={index} isPercent={this.props.indicator.isPercent} {...values} />
            );
        } else {
            return this.props.indicator.indicatorData.map(
                (value, index) => {
                    return <td key={index} align="right">
                        {value ? String(Math.round(value)) + (this.props.indicator.isPercent ? '%' : '') : EMPTY_CELL}</td>;
                }
            );
        }
    }
    render() {
        let indicator = this.props.indicator;
        if (indicator.id === null) {
            return <Loading />;
        }
        let resultsButton = <button type="button" className="btn btn-link p-1 indicator-ajax-popup indicator-data"
                         data-indicatorid={indicator.id}
                         data-container="body"
                         data-trigger="focus"
                         data-toggle="popover"
                         data-placement="bottom">
                            <i className="fas fa-table"></i>
                        </button>;
        let updateButton = <a href="#" className="indicator-link float-right">
                                <i className="fas fa-cog"></i>
                            </a>;
        let baseline = indicator.baseline ? String(indicator.baseline) + (indicator.isPercent ? '%' : '') : EMPTY_CELL;
        let lopTarget = indicator.lopTarget ? String(Math.round(indicator.lopTarget)) + (indicator.isPercent ? '%' : '') : EMPTY_CELL;
        let lopActual = indicator.lopActual
                        ? indicator.isPercent
                            ? String(Math.round(indicator.lopActual * 10)/10) + '%'
                            : String(Math.round(indicator.lopActual))
                        : EMPTY_CELL;
        let lopMet = indicator.lopMet ? String(Math.round(indicator.lopMet * 1000)/10) + '%' : EMPTY_CELL;
        return <tr>
                    <IndicatorTD>{ indicator.number }</IndicatorTD>
                    <IndicatorTD>{ resultsButton }  { indicator.name }</IndicatorTD>
                    <IndicatorTD>{ updateButton }</IndicatorTD>
                    <IndicatorTD>{ indicator.level }</IndicatorTD>
                    <IndicatorTD>{ indicator.unitOfMeasure }</IndicatorTD>
                    <IndicatorTD align="right">{ indicator.directionOfChange }</IndicatorTD>
                    <IndicatorTD>{ indicator.cumulative }</IndicatorTD>
                    <IndicatorTD>{ indicator.unitType }</IndicatorTD>
                    <IndicatorTD align="right">{ baseline }</IndicatorTD>
                    <IndicatorTD align="right">{ lopTarget }</IndicatorTD>
                    <IndicatorTD align="right">{ lopActual }</IndicatorTD>
                    <IndicatorTD align="right">{ lopMet }</IndicatorTD>
                    { this.getPeriodValues() }
               </tr>;
    }
}

const Loading = () => {
    return <tr><td>Loading</td></tr>;
}

const NoIndicatorsForFrequency = inject('labels')(({ labels }) => {
    return <tr><td colSpan="8">{ labels.noIndicatorsForFrequency }</td></tr>;
})

const IPTTTableBody = inject('rootStore')(
    observer(
        ({rootStore}) => {
            const indicatorRows = (rootStore.reportIndicators && rootStore.reportIndicators.length > 0)
                                  ? rootStore.reportIndicators.map(
                                    (indicator, count) => <IndicatorRow indicator={indicator} key={count} />
                                  ) : rootStore.noIndicatorsForFrequency
                                    ? <NoIndicatorsForFrequency />
                                    : <Loading />;
            
            return <tbody>
                        {indicatorRows}
                   </tbody>
        }
    )
)

@inject('labels', 'rootStore')
@observer
export class IPTTTable extends React.Component {
    render() {
        return <table className="table table-sm table-bordered table-hover" id="iptt_table">
                    <colgroup scope="col" span="9"></colgroup>
                    <colgroup scope="col" span="3" className="lopCols"></colgroup>
                    <colgroup scope="col" span={ this.props.rootStore.selectedPeriods.length }></colgroup>
                    <IPTTTableHead />
                    <IPTTTableBody />
               </table>;
    }
}