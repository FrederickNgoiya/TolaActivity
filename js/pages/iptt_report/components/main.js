import React from 'react';
import { inject, observer } from 'mobx-react';
import { IPTTTable } from './reportComponents';
import { IPTTFilterForm } from './filterForm';
import { PinButton, ExcelButton } from './buttons';


const IPTTHeader = inject('labels', 'rootStore')(
    observer(({ labels, rootStore }) => {
        return <div className="page-subheader">
                    <div id="id_span_iptt_date_range" className="subheader__title">
                        <h2 className="pt-3 text-title-case">{ labels.reportTitle }</h2>
                        <h4 className="pb-3">{ (rootStore.startPeriodLabel && rootStore.endPeriodLabel)
                                               ? rootStore.startPeriodLabel + " - " + rootStore.endPeriodLabel
                                               : "" }</h4>
                    </div>
                    <div className="subheader__actions">
                        <PinButton />
                        <ExcelButton />
                    </div>
                </div>
    })
);


const IPTTSidebar = inject('labels')(({ labels }) => {
    return <div className="sidebar_wrapper">
                <div className="collapse width show" id="sidebar">
                  <IPTTFilterForm />
                </div>
                <div className="sidebar-toggle">
                  <a href="#" data-target="#sidebar" data-toggle="collapse"
                        title={ labels.sidebarToggle }>
                    <i className="fa fa-chevron-left"></i>
                  </a>
                </div>
            </div>;
});


const IPTTReport = () => {
    return <main className="iptt_table_wrapper">
                <div id="id_div_top_iptt_report">
                    <IPTTHeader />
                    <IPTTTable />
                </div>
            </main>;
                
}

export class IPTTReportApp extends React.Component {
    render() {
        return <React.Fragment>
            <IPTTSidebar />
            <IPTTReport />
        </React.Fragment>
    }
}