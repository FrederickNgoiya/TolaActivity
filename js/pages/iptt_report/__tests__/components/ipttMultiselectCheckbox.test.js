import React from 'react';
import renderer from 'react-test-renderer';
import { IPTTMultiselectCheckboxWrapper } from '../../components/selectWidgets';


const labels = {
    emptySelect: 'empty label',
    noOptionsSelect: 'no options label',
    selected: 'selected'
};

class TestItem extends IPTTMultiselectCheckboxWrapper {
    onChange = (selected) => {
        this.props.testHarness.selectedValue = selected;
    }
    get value() {
        return this.props.testHarness.selectedValue;
    }
    get options() {
        return this.props.testHarness.options;
    }
    get label() {
        return 'label';
    }
}

describe('MultiselectCheckbox', () => {
    it('displays an empty select label with nothing selected but options', () => {
        let testHarness = {
            options: [{label: 'option 1', value: 1}, {label: 'option 2', value: 2}],
            selectedValue: []
        };
        const component = renderer.create(<TestItem labels={labels} testHarness={testHarness} />);
        let nothingSelected = component.toJSON();
        expect(nothingSelected).toMatchSnapshot();
    });
})