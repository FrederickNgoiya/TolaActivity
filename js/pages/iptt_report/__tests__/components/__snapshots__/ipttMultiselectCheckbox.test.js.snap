// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiselectCheckbox displays an empty select label with nothing selected but options 1`] = `
<div
  className="form-row mb-2 iptt-react-select-row"
>
  <label
    className="col-form-label text-uppercase"
    htmlFor="multiselect1"
  >
    label
  </label>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <button
      className="css-1r4vtzz"
      onClick={[Function]}
      type="button"
    >
      <span
        className="css-1v99tuv"
      >
        empty label
      </span>
      <span
        className="css-1gpjby2"
      >
        <svg
          focusable="false"
          height="24"
          role="presentation"
          style={
            Object {
              "marginRight": -6,
            }
          }
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M8.292 10.293a1.009 1.009 0 0 0 0 1.419l2.939 2.965c.218.215.5.322.779.322s.556-.107.769-.322l2.93-2.955a1.01 1.01 0 0 0 0-1.419.987.987 0 0 0-1.406 0l-2.298 2.317-2.307-2.327a.99.99 0 0 0-1.406 0z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
